File: src/main/java/dev/silverandro/hurry_up/PlayerTimer.java
package dev.silverandro.hurry_up;

public interface PlayerTimer {
	void hurry_up$setDefused(boolean isDefused);
	void hurry_up$setTicks(int count);
	void hurry_up$setlastDeathTime(long time);

	boolean hurry_up$isDefused();
	int hurry_up$getRemainingTicks();
	long hurry_up$getLastDeathTime();
}

File: src/main/kotlin/dev/silverandro/hurry_up/util/PlayerTimer.kt
package dev.silverandro.hurry_up.util

import dev.silverandro.hurry_up.PlayerTimer
import net.minecraft.entity.player.PlayerEntity

var PlayerEntity.defused: Boolean
    get() = (this as PlayerTimer).`hurry_up$isDefused`()
    set(value) = (this as PlayerTimer).`hurry_up$setDefused`(value)

var PlayerEntity.remainingTicks: Int
    get() = (this as PlayerTimer).`hurry_up$getRemainingTicks`()
    set(value) = (this as PlayerTimer).`hurry_up$setTicks`(value)

var PlayerEntity.lastDeathTime: Long
    get() = (this as PlayerTimer).`hurry_up$getLastDeathTime`()
    set(value) = (this as PlayerTimer).`hurry_up$setlastDeathTime`(value)
